tests/Backend/ReaderWorkspaces/WorkspacesReaderSharingTest.php

```diff
@@ -0,0 +1,238 @@
+                $components->deleteConfiguration($component['id'], $configuration['id']);
+            }
+        }
+
+        // erase all deleted configurations
+        foreach ($components->listComponents((new ListComponentsOptions())->setIsDeleted(true)) as $component) {
+            foreach ($component['configurations'] as $configuration) {
+                $components->deleteConfiguration($component['id'], $configuration['id']);
+            }
+        }
+
+        if (self::$client === null) {
+            self::$client = $this->_client;
+        }
+
+        if (self::$backendConnection === null) {
+            self::$backendConnection = $this->ensureSnowflakeConnection();
+        }
+
+        // drop all reader workspaces before test
+        $workspaces = new Workspaces(self::$client);
+        $filteredReaderWorkspaces = array_filter($workspaces->listWorkspaces(), static fn($workspace
+        ) => $workspace['platformUsageType'] === 'reader');
+        foreach ($filteredReaderWorkspaces as $workspace) {
+            $workspaces->deleteWorkspace($workspace['id']);
+        }
+
+        // drop all readers accounts
+        self::dropReaderAccounts();
+
+        // ensure the workspace and the reader account is cleaned up.
+        self::ensureReaderAccountIsRemoved();
+
+        // unlink buckets
+        $linkedBuckets = $linkClient->listBuckets();
+        foreach ($linkedBuckets as $bucket) {
+            $linkClient->dropBucket($bucket['id'], ['force' => true]);
+        }
+
+        $shareClient = $this->getClient([
+            'token' => STORAGE_API_SHARE_TOKEN,
+            'url' => STORAGE_API_URL,
+        ]);
+
+        foreach ($shareClient->listBuckets() as $bucket) {
+            if ($shareClient->isSharedBucket($bucket['id'])) {
+                $shareClient->unshareBucket($bucket['id']);
+            }
+        }
+
+        $this->_initEmptyTestBuckets();
+    }
+
+    public function testShareLinkAndLoadTableToReaderWorkspace(): void
+    {
+        $linkClient = $this->getClient([
+            'token' => STORAGE_API_LINKING_TOKEN,
+            'url' => STORAGE_API_URL,
+        ]);
+
+        // Step 1: Create a source bucket and table
+        $this->_client->createTableAsync(
+            $this->getTestBucketId(),
+            'languages',
+            new CsvFile(__DIR__ . '/../../_data/languages.csv'),
+        );
+
+        $shareClient = $this->getClient([
+            'token' => STORAGE_API_SHARE_TOKEN,
+            'url' => STORAGE_API_URL,
+        ]);
+
+        $tokenInfo = $linkClient->verifyToken();
+        $projectId = $tokenInfo['owner']['id'];
+        // Step 2: Share the bucket with the organization
+        $shareClient->shareBucketToProjects($this->getTestBucketId(), [$projectId]);
+
+        // Step 3: Link the shared bucket to the current project
+        $response = $linkClient->listSharedBuckets();
+        $this->assertCount(1, $response);
```

Comment:
toto nebude stabilní měl bych ověřit že je nasharovaný ten co chceš ne že je tam jeden. někdo přidá další test do suity a začne to padat.